### ISSUE TITLE
Conflict during ocaml-lsp-server installation

### ISSUE THREAD

**Issue by Gabyfle** [ocaml/ocaml-lsp#149]
I'm getting this error using `opam install ocaml-lsp-server` :

 --debug
```
gabyfle@Dellinux:~$ opam install ocaml-lsp-server --debug
00:00.002  GSTATE                 LOAD-GLOBAL-STATE @ /home/gabyfle/.opam
00:00.002  RSTATE                 LOAD-REPOSITORY-STATE @ /home/gabyfle/.opam
00:00.083  RSTATE                 Loaded /home/gabyfle/.opam/repo/state.cache in 0.081s
00:00.124  RSTATE                 Cache found
00:00.124  STATE                  LOAD-SWITCH-STATE @ default
00:00.174  STATE                  Detected changed packages (marked for reinstall): {}
00:00.188  STATE                  Switch state loaded in 0.064s
00:00.189  FILE(switch-state)     Wrote /home/gabyfle/.opam/default/.opam-switch/backup/state-20200412100248.export in 0.000s

<><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
00:00.306  UPDATE                 update-dev-packages
00:00.306  PARALLEL               Iterate over 1 task(s) with 3 process(es)
00:00.306  PARALLEL               Starting job 0 (worker 1/3): 0
00:00.306  UPDATE                 update-dev-package ocaml-lsp-server.~dev
00:00.306  UPDATE                 update-pinned-package ocaml-lsp-server
00:00.308  UPDATE                 updating git+https://github.com/ocaml/ocaml-lsp.git
00:00.312  PARALLEL               Next task in job 0: /usr/bin/git remote set-url origin https://github.com/ocaml/ocaml-lsp.git
Processing  1/1: [ocaml-lsp-server.~dev: git]
00:00.316  PARALLEL               Collected task for job 0 (ret:0)
00:00.321  PARALLEL               Next task in job 0: /usr/bin/git fetch -q https://github.com/ocaml/ocaml-lsp.git --update-shallow +HEAD:refs/remotes/opam-ref
00:00.970  PARALLEL               Collected task for job 0 (ret:0)
00:00.975  PARALLEL               Next task in job 0: /usr/bin/git push /home/gabyfle/.opam/download-cache/git +refs/remotes/opam-ref:refs/remotes/b91a202b4c9ba3c0bd3e92f80993b257
00:00.986  PARALLEL               Collected task for job 0 (ret:0)
00:00.991  PARALLEL               Next task in job 0: /usr/bin/git diff --no-ext-diff --quiet refs/remotes/opam-ref --
00:01.016  PARALLEL               Collected task for job 0 (ret:0)
[ocaml-lsp-server.~dev] no changes from git+https://github.com/ocaml/ocaml-lsp.git
00:01.017  PARALLEL               Job 0 finished
00:01.017  SWACT                  add-to-reinstall unpinned_only:false packages:{}

00:01.017  CLIENT                 INSTALL ocaml-lsp-server
00:01.270  SYSTEM                 [log-15878-05296d] (in 0.042s) lsb_release -s -r
00:01.292  SYSTEM                 [log-15878-d2c37b] (in 0.012s) ocamlc -vnum
00:01.437  CLIENT                 Orphans: (changes: { ocaml-lsp-server.~dev }, transitive: false) -> full {}, versions {}
00:01.754  SOLVER                 resolve request=install:(ocaml-lsp-server (= ~dev)) remove:() upgrade:()
00:02.701  SOLVER                 Load cudf universe (depopts:false, build:true, post:true)
00:02.805  CUDF                   resolve request=install:(ocaml-lsp-server (= 1)) remove:() upgrade:()
00:02.805  SOLVER                 Calling solver builtin-mccs+glpk with criteria -removed,-count[version-lag,request],-count[version-lag,changed],-changed
00:03.626  CUDF                   Solver call done in 0.821
00:04.319  CLIENT                 conflict!
Sorry, no solution found: there seems to be a problem with your request.

No solution found, exiting
00:04.319  SYSTEM                 rm /home/gabyfle/.opam/default/.opam-switch/backup/state-20200412100248.export
``` 

The only error message I get is : `conflict!`  which isn't very accurate...

If anyone has any idea how I can fix this, I'll take it!

**Comment by rgrinberg** [comment]
What command are you using to pin? What version of OCaml do you have?

Did you run opam update?

**Comment by Gabyfle** [comment]
Hi !
I didn't run ` opam update` (I did now, but still not fixed)
`ocaml -version` returns ` 4.05.0`
To pin this repository I'm using the given command in the Opam section of the README file which is : 

` opam pin add ocaml-lsp-server https://github.com/ocaml/ocaml-lsp.git` 

The problem is that I don't have that much information on where does the problem comes from. I don't know if this is a known issue but actually, `opam` isn't that precise on the errors

**Comment by rgrinberg** [comment]
You need a more recent version of OCaml. This Lsp server needs 4.06 or above

**Comment by Gabyfle** [comment]
> You need a more recent version of OCaml. This Lsp server needs 4.06 or above
> […](#)
> On Apr 13, 2020, 9:56 AM -0700, Gabriel Santamaria ***@***.***>, wrote: Hi ! I didn't run opam update (I did now, but still not fixed) ocaml -version returns 4.05.0 To pin this repository I'm using the given command in the Opam section of the README file which is : opam pin add ocaml-lsp-server https://github.com/ocaml/ocaml-lsp.git The problem is that I don't have that much information on where does the problem comes from. I don't know if this is a known issue but actually, opam isn't that precise on the errors — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

Ho craps I didn't see that it requires OCaml 4.06. Actually maybe you should add a "Requirements" section in README ?

Thanks for quick answer, closing.
